server/src/test/java/org/elasticsearch/snapshots/SnapshotsServiceTests.java

```diff
@@ -0,0 +1,526 @@
+                                throw new AssertionError(e);
+                            }
+                        }
+                    ).when(channel).sendResponse(any(TransportResponse.class));
+                    handlerRegistry.processMessageReceived(request, channel);
+                } catch (Exception e) {
+                    throw new AssertionError(e);
+                }
+            }
+        };
+
+        TestClusterNode(DiscoveryNode node, DeterministicTaskQueue deterministicTaskQueue) throws IOException {
+            this.node = node;
+            masterService = new FakeThreadPoolMasterService(node.getName(), "test", deterministicTaskQueue::scheduleNow);
+            final Settings settings = Settings.builder()
+                .put(NODE_NAME_SETTING.getKey(), node.getName())
+                .put(PATH_HOME_SETTING.getKey(), createTempDir().toAbsolutePath().toString())
+                .build();
+            allocationService = ESAllocationTestCase.createAllocationService(settings);
+            final ClusterSettings clusterSettings = new ClusterSettings(settings, ClusterSettings.BUILT_IN_CLUSTER_SETTINGS);
+            final ThreadPool threadPool = deterministicTaskQueue.getThreadPool();
+            clusterService = new ClusterService(settings, clusterSettings, threadPool, masterService);
+            this.deterministicTaskQueue = deterministicTaskQueue;
+            transportService = mockTransport.createTransportService(
+                settings, deterministicTaskQueue.getThreadPool(
+                    runnable -> () -> {
+                        try (CloseableThreadContext.Instance ignored =
+                                 CloseableThreadContext.put("nodeId", '{' + node.getId() + "}{" + node.getEphemeralId() + '}')) {
+                            runnable.run();
+                        }
+                    }
+                ), NOOP_TRANSPORT_INTERCEPTOR,
+                a -> node, null, emptySet()
+            );
+            final IndexNameExpressionResolver indexNameExpressionResolver = new IndexNameExpressionResolver();
+            snapshotsService =
+                new SnapshotsService(settings, clusterService, indexNameExpressionResolver, repositoriesService, threadPool);
+            final Environment environment = new Environment(settings, createTempDir());
+            nodeEnv = new NodeEnvironment(settings, environment);
+            final NamedXContentRegistry namedXContentRegistry = new NamedXContentRegistry(Collections.emptyList());
+            final ScriptService scriptService = new ScriptService(settings, emptyMap(), emptyMap());
+            client = new NodeClient(settings, threadPool);
+            final IndexScopedSettings indexScopedSettings =
+                new IndexScopedSettings(settings, IndexScopedSettings.BUILT_IN_INDEX_SETTINGS);
+            indicesService = new IndicesService(
+                settings,
+                mock(PluginsService.class),
```

Comment:
Just mocked this one out since it's not relevant for the test and just a bunch of code to get up and running.